components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImpl.java

```diff
@@ -140,6 +143,28 @@ public Action getActionByActionId(String actionId, String tenantDomain) throws A
         return CACHE_BACKED_DAO.getActionByActionId(actionId, IdentityTenantUtil.getTenantId(tenantDomain));
     }
 
+    @Override
+    public Action updateActionEndpointAuthentication(String actionType, String actionId, AuthType authentication,
+                                                     String tenantDomain) throws ActionMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Updating Action endpoint authentication for Auth Type: " + authentication.getType().name() +
+                    " and Action Type: " + actionType + " and Action Id: " + actionId);
+        }
+        Action action = checkIfActionExists(actionId, tenantDomain);
+        if (action.getEndpoint().getAuthentication().getType().equals(authentication.getType())) {
+
+            return CACHE_BACKED_DAO.updateActionEndpointAuthProperties(actionId, authentication,
+                    IdentityTenantUtil.getTenantId(tenantDomain));
+        } else {
+            EndpointConfig endpoint = new EndpointConfig.EndpointConfigBuilder()
+                    .uri(action.getEndpoint().getUri())
+                    .authentication(authentication).build();
+            return CACHE_BACKED_DAO.updateActionEndpoint(getActionTypeFromPath(actionType), actionId, endpoint,
```

Comment:
Doesn't the Action retrieved above include the action type ? Is retrieving action type needed ?

Reply:
Moreover, this function just change the auth type reference in the action table. Why can't we just update that rather than updating the action object fully

Reply:
`actionType` is required to invalidate the cache in the caching layer.
And here we are not only updating the authType. We are updating the authType and the respective authProperties in the table while deleting the old auth properties.